apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPoolOptions.cs

```diff
@@ -22,6 +23,24 @@ namespace Google.Cloud.Spanner.V1
     public sealed class SessionPoolOptions
     {
         /// <summary>
+        /// Constructs a new <see cref="SessionPoolOptions"/>.
+        /// </summary>
+        public SessionPoolOptions()
+        {
+            var retry = SpannerSettings.GetDefault().CommitSettings.Timing.Retry;
+            var timeout = retry?.TotalExpiration.Timeout;
+            if (timeout.HasValue)
+            {
+                Timeout = (int)timeout.Value.TotalSeconds;
+            }
+            else
+            {
+                Debug.Assert(false, "Config change has invalidated default timeout calc");
+                Timeout = (int)TimeSpan.FromMilliseconds(36000000).TotalSeconds;
```

Comment:
10 hours?

Reply:
that's whats in gapic:

https://github.com/googleapis/googleapis/pull/424/files#diff-4501db3e3507bcce8496c9aea2017d96R54

bug there?

Reply:
Oops. That is bug in there. Let me fix it first.

Reply:
https://github.com/googleapis/googleapis/pull/438 sent

Reply:
thanks.  I'll update this PR with a code regen after that's merged.

Reply:
It's not clear to me why we need this "else" part. If it's guarding against the settings having been changed in the YAML file incorrectly, won't the unit test catch that? We wouldn't release something without running the unit tests, so shouldn't that be okay?

I'd prefer for us to simply assume everything's okay in the production code, at which point you could get rid of the constructor entirely and just use a property initializer:

```csharp
public int Timeout { get; set; } = (int) SpannerSettings.GetDefault().CommitSettings.Timing.Retry.TotalExpiration.Timeout.TotalSeconds;
```

Reply:
it's actually a bit difficult to debug when this changes underneath you.  I initially used a property initializer, got the initialization wrong and proceeded to get a whole bunch of typeloadexceptions from the static initializer of sessionpooloptions which indirectly creates once of these in its cctor.

If you *really* feel strong about this I'll consider changing, but after that experience, I'd rather it fail in a very clear and diagnosable way.

Reply:
Happy to have it in the constructor for easier debugging. (It will still fail in the cctor of SessionPoolOptions if it's in the constructor here, of course... but it may be easier to break on.)

I think I'd still rather not try to "recover" from it being broken though, with the else block you've currently got. Does that make sense?

Reply:
It should never happen in a released build, so I'm fine with that.  I've been waiting for gapic config to be updated so I can regen the libs -- I'll update this change with that (it will have 2 separate commits).

Reply:
Regen is now merged, so assuming it was the one you were waiting for, it should be simple after a rebase.